src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -710,6 +712,45 @@
          :headers {"Content-Type" "application/octet-stream"}
          :body (get-code-content environment code-id file-path)}))))
 
+(defn decode-catalog-post-body
+  [body]
+  (let [parameters
+        (try+
+          (json/decode body true)
+          (catch JsonParseException e
+            (throw+ {:kind :bad-request
+                     :msg (format "Error parsing JSON: %s" e)})))]
+    (schema/validate
+     {:certname schema/Str
+      :persistence {:facts schema/Str, :catalogs schema/Str, :reports schema/Str}
+      (schema/optional-key :trusted_facts) {:values {schema/Any schema/Any}}
+      (schema/optional-key :facts) {:values {schema/Any schema/Any}}
+      (schema/optional-key :job-id) schema/Int
```

Comment:
Can we choose either hyphen or underscores for these keys (this vs. `trusted_facts` above)? I don't have a strong preference which we choose.

Reply:
@tvpartytonight where does this job-id come from?

Reply:
looks like just an error, i've fixed it to _